libcontainer/cgroups/fs/cpuacct.go

```diff
@@ -91,15 +91,29 @@ func getCpuUsageBreakdown(path string) (uint64, uint64, error) {
 	if err != nil {
 		return 0, 0, err
 	}
-	// TODO: use strings.SplitN instead.
-	fields := strings.Fields(data)
-	if len(fields) < 4 || fields[0] != userField || fields[2] != systemField {
+
+	userLine, systemLine, found := strings.Cut(data, "\n")
+	if !found {
+		return 0, 0, malformedLine(path, file, data)
+	}
+
+	userFieldWord, userUsageInTicks, found := strings.Cut(userLine, " ")
+	if !found || userFieldWord != userField {
 		return 0, 0, malformedLine(path, file, data)
 	}
-	if userModeUsage, err = strconv.ParseUint(fields[1], 10, 64); err != nil {
+
+	userUsageInTicks = strings.TrimSpace(userUsageInTicks)
```

Comment:
Are there extra spaces?

Reply:
When I implemented the change with `strings.SplitN`, I received the following error from the tests, which is why I added the `userUsageInTicks = strings.TrimSpace(userUsageInTicks)` that fixed this error.

```
# time="2024-09-15T14:08:22Z" level=error msg="unable to get container cgroup stats:
unable to parse /sys/fs/cgroup/cpu,cpuacct/system.slice/runner-provisioner.service/test_busybox/cpuacct.stat:
strconv.ParseUint: parsing \"1\\n\": invalid syntax"
```

Reply:
Ah fwiw that's probably because `lines := strings.SplitN(data, "\n", 2)` would only "eat" the middle newline where the file contents have two newlines (`line1\nline2\n`) and `strings.Fields` splits on _any_ whitespace, including consuming consecutive and anything extra at the beginning and end, hence exactly 4 fields.

Reply:
Yes, `strings.Fields` removes all whitespace around which is ideal for this use case.